src/CLaSH/Prelude/Explicit/Safe.hs

```diff
@@ -114,3 +118,23 @@ isFalling' clk is s = liftA2 edgeDetect prev s
   where
     prev = register' clk is s
     edgeDetect old new = old == maxBound && new == minBound
+
+{-# INLINEABLE riseEvery' #-}
+-- | Give a pulse every @n@ clock cycles. This is a useful helper function when
+-- combined with functions like @'CLaSH.Signal.regEn'@ or @'CLaSH.Signal.mux'@,
+-- in order to delay a register by a known amount.
+riseEvery' :: KnownNat n => SClock clk -> Unsigned n -> Signal' clk Bool
+riseEvery' clk cnt = moore' clk transfer output 1 (pure ())
+  where
+    output   s   = if (s == cnt) then True else False
+    transfer s _ = if (s == cnt) then 1    else s+1
+
+{-# INLINEABLE oscillate' #-}
+-- | Oscillate a @'Bool'@ for a given number of cycles, given the starting state.
+oscillate' :: KnownNat n => SClock clk -> Bool -> Unsigned n -> Signal' clk Bool
```

Comment:
@thoughtpolice I think I prefer `SNat n` over `Unsigned n`, as you can never change the count dynamically (`Signal` doesn't have a `Monad` instance). Internally you would then use `Index n` for the counter. You would then get:

```haskell
oscillate' :: forall clk n . SClock clk -> Bool -> SNat n -> Signal' clk Bool
oscillate' clk begin cnt@SNat = moore' clk transfer snd (0, begin) (pure ())
  where
    transfer :: (Index n, Bool) -> () -> (Index n, Bool)
    transfer (s, i) _ =
      if s < snatToNum cnt
        then (s + 1,i)  -- hold current output
        else (1, not i) -- reset state and oscillate output
```

I don't think we should care too much about the probability that `< cnt` might use more logic than `== cnt`

A similar change would then have to be applied to `riseEvery'`